website/content/docs/concepts/integrated-storage.mdx

```diff
@@ -169,15 +179,117 @@ node2    node2.vault.local:8201    follower    true
+replication of stored data.
+
+The cluster port is secured using a TLS certificate that the Vault active node
+generates at startup.  It's clear how this can work when not using integrated
+storage: every node has at least read access to storage, so once the active
+node has persisted the certificate, the standby nodes can fetch it, and all
+agree on how cluster traffic should be encrypted.
+
+It's less clear how this works with integrated storage, as there is a chicken
+and egg problem.  Nodes don't have a shared view of storage until the raft
+cluster has been formed, but we're trying to form the raft cluster!  To solve
+this problem, a Vault node must speak to another Vault node using the API port
+instead of the cluster port.  This is currently the only situation in which
+OSS Vault does this (Vault Enterprise also does something similar when setting
+up replication.)
+
+* node2 wants to join the cluster, so issues challenge API request to existing member node1
+* node1 replies to challenge request with (1) an encrypted random UUID and (2) seal config
+* node2 must decrypt UUID using seal; if using auto-unseal can do it directly, if using shamir must wait for user to provide enough unseal keys to perform decryption
+* node2 sends decrypted UUID back to node1 using answer API
+* node1 sees node2 can be trusted (since it has seal access) and replies with a bootstrap package which includes the cluster TLS certificate and private key
+* node2 gets sent a raft snapshot over the cluster port
+
+After this procedure the new node will never again send traffic to the API port.
+All subsequent inter-node communication will use the cluster port.
+
+![Raft Join Process](/img/raft-join-detailed.png)
+
+### Assisted raft join techniques
+
+The simplest but least desirable solution is to do it by hand: issue raft join
+commands specifying the explicit names or IPs of the nodes to join to.  In this
+section we look at other TLS-compatible options that lend themselves more to automation.
+
+#### Autojoin with TLS servername
+
+As of Vault 1.6.2, the simplest option might be to specify a leader_tls_servername
+in the retry_join stanza which matches a DNS SAN in the certificate.
+
+Note that names in a certificate's DNS SAN don't actually have to be registered
+in a DNS server.  Your nodes may have no names found in DNS, while still
+using certificate(s) that contain this shared "servername" in their DNS SANs.
+
+#### Autojoin but constrain CIDR, list all possible IPs in certificate
+
+If all the vault node IPs are assigned from a small subnet, e.g. a /28, it
+becomes practical to put all the IPs that exist in that subnet into the IP SANs
+of the TLS certificate the nodes will share.
+
+The drawback here is that the cluster may someday outgrow the CIDR and changing
+it may be a pain.  For similar reasons this solution may be impractical when
+using non-voting nodes and dynamically scaling clusters.
+
+#### Autojoin using a second TCP listener
```

Comment:
This one is insecure, let's not even mention it as an option

Reply:
Removed.